tests/catch/unit/memory/hipFree.cc

```diff
@@ -0,0 +1,522 @@
+
+
+  size_t width = GENERATE(32, 512, 1024);
+  size_t height = GENERATE(32, 512, 1024);
+
+  SECTION("ArrayFree") {
+    hipArray_t arrayPtr{};
+    hipChannelFormatDesc desc = hipCreateChannelDesc<TestType>();
+
+    HIP_CHECK(hipMallocArray(&arrayPtr, &desc, width, height, hipArrayDefault));
+    runKernelForMs(50);
+    // make sure device is busy
+    HIP_CHECK_ERROR(hipStreamQuery(nullptr), hipErrorNotReady);
+    HIP_CHECK(hipFreeArray(arrayPtr));
+    HIP_CHECK(hipStreamQuery(nullptr));
+  }
+  SECTION("ArrayDestroy") {
+    hiparray cuArrayPtr{};
+
+    HIP_ARRAY_DESCRIPTOR cuDesc;
+    cuDesc.Width = width;
+    cuDesc.Height = height;
+    cuDesc.Format = vec_info::format;
+    cuDesc.NumChannels = vec_info::size;
+    HIP_CHECK(hipArrayCreate(&cuArrayPtr, &cuDesc));
+    runKernelForMs(50);
+    // make sure device is busy
+    HIP_CHECK_ERROR(hipStreamQuery(nullptr), hipErrorNotReady);
+    HIP_CHECK(hipArrayDestroy(cuArrayPtr));
+    HIP_CHECK(hipStreamQuery(nullptr));
+  }
+}
+#else  // AMD
+
+TEMPLATE_TEST_CASE("Unit_hipFreeImplicitSyncArray", "", char, float, float2, float4) {
+  hipArray_t arrayPtr{};
+  hipExtent extent{};
+  extent.width = GENERATE(32, 128, 256, 512, 1024);
+  extent.height = GENERATE(0, 32, 128, 256, 512, 1024);
+  hipChannelFormatDesc desc = hipCreateChannelDesc<TestType>();
+
+  HIP_CHECK(hipMallocArray(&arrayPtr, &desc, extent.width, extent.height, hipArrayDefault));
+  runKernelForMs(50);
+  // make sure device is busy
+  HIP_CHECK_ERROR(hipStreamQuery(nullptr), hipErrorNotReady);
+  // Second free segfaults
+  SECTION("ArrayDestroy") {
+    HIP_CHECK(hipArrayDestroy(arrayPtr));
+    HIP_CHECK(hipStreamQuery(nullptr));
+  }
+  SECTION("ArrayFree") {
+    HIP_CHECK(hipFreeArray(arrayPtr));
+    HIP_CHECK(hipStreamQuery(nullptr));
+  }
+}
+
+#endif
+
+// Freeing a invalid pointer with on device
+TEST_CASE("Unit_hipFreeNegativeDev") {
+  SECTION("InvalidPtr") {
+    char value;
+    HIP_CHECK_ERROR(hipFree(&value), hipErrorInvalidValue);
+  }
+  SECTION("NullPtr") { HIP_CHECK(hipFree(nullptr)); }
+}
+
+// Freeing a invalid pointer with on device
+TEST_CASE("Unit_hipFreeNegativeHost") {
+  char* hostPtr{nullptr};
+  SECTION("NullPtr") { HIP_CHECK(hipHostFree(nullptr)); }
+  SECTION("InvalidPtr") {
+    hostPtr = new (char);
+    HIP_CHECK_ERROR(hipHostFree(hostPtr), hipErrorInvalidValue);
+  }
+  SECTION("hipHostRegister") {
+    hostPtr = new char;
+    auto flag = GENERATE(hipHostRegisterDefault, hipHostRegisterPortable, hipHostRegisterMapped);
+    HIP_CHECK(hipHostRegister((void*)hostPtr, sizeof(char), flag));
+    HIP_CHECK_ERROR(hipHostFree(hostPtr), hipErrorInvalidValue);
+  }
+  delete hostPtr;
+}
+
+#if HT_NVIDIA
+TEST_CASE("Unit_hipFreeNegativeArray") {
+  DriverContext ctx;
+  hipArray_t arrayPtr{};
+  hiparray cuArrayPtr{};
+
+  SECTION("InvalidPtr") {
+    arrayPtr = static_cast<hipArray*>(malloc(sizeof(char)));
+    cuArrayPtr = static_cast<hiparray>(malloc(sizeof(char)));
```

Comment:
```suggestion
    arrayPtr = static_cast<hipArray*>(0xDEAD);
    cuArrayPtr = static_cast<hiparray>(0xBEEF);
```
if you just want so invalid values to tests as pointer.
Personally I think this test is unnecessary and too much like undefined behaviour. Also I think the behaviour is covered in the double free test.

Reply:
Considered it, and I agree double free captures the case well enough invalid ptr sections removed